pkg/build/graph/edges.go

```diff
@@ -22,22 +21,24 @@ const (
 
 // AddInputOutputEdges links the build config to other nodes for the images and source repositories it depends on.
 func AddInputOutputEdges(g osgraph.MutableUniqueGraph, node *buildgraph.BuildConfigNode) *buildgraph.BuildConfigNode {
-	output := node.BuildConfig.Parameters.Output
+	output := node.BuildConfig.Spec.Output
 	to := output.To
 	switch {
-	case to != nil && len(to.Name) > 0:
-		out := imagegraph.FindOrCreateSyntheticImageStreamTagNode(g, imagegraph.MakeImageStreamTagObjectMeta(defaultNamespace(to.Namespace, node.BuildConfig.Namespace), to.Name, output.Tag))
+	case to == nil:
+	case to.Kind == "DockerImage":
+		out := imagegraph.EnsureDockerRepositoryNode(g, to.Name, "")
```

Comment:
the tag is part of the name in those cases, if there is one, so ParseDockerImageReference takes care of it.